dask_expr/_expr.py

```diff
@@ -1275,6 +1276,196 @@ def _task(self, index: int):
             )
 
 
+class MapOverlap(MapPartitions):
```

Comment:
Mostly a "note to self": I'm wondering if there is any "danger" in treating the pre-lowered version of `MapOverlap` as a `Blockwise` expression. It is very important that we don't do any partition-related optimizations (e.g. culling) until after this expression is lowered. It seems like we are in the clear for the current `optimize` implementation, but it isn't clear to me that we are gaining much by inheriting from `MapPartitions`  to begin with?

Reply:
Culling in general has to happen after lower, but I agree that this isn't ideal.

Maybe we have to create a new class that sits somewhere in between Expr and Blockwise, but removing this information removes a lot of information that is helpful